### PR TITLE
feat: fully reprint AST with @babel/generator

### LINKED ISSUE #179: 🐛 Bug: comments are removed on prettier run
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/prettier-plugin-curly/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

"prettier-plugin-curly": "0.1.3":

This code...
```ts
if(document.body)
  //Important comment which will be lost!
  document.body.append('')
```
should be transformed to
```ts
if(document.body) {
  //Important comment which will be lost!
  document.body.append('')
}
```

### Actual

I get
```ts
if(document.body) {
  document.body.append('')
}
```

### Additional Info

Same with block comments: `/* comment */`

### LINKED ISSUE #284: 🐛 Bug: prettier-plugin-curly crashes on nested if statements
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/prettier-plugin-curly/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

I expected to be able to run `pnpm format` with `prettier-plugin-curly` on literate-ts.

### Actual

```
$ npx prettier src/ts-checker.ts
src/ts-checker.ts
[error] src/ts-checker.ts: SyntaxError: ',' expected. (85:5)
[error]   83 |       const start = m.index + 1;
[error]   84 |       const end = start + m[1].length;
[error] > 85 |       errors.push({line: lastCodeLine, start, end, message});
[error]      |     ^
[error]   86 |     }
[error]   87 |   });
[error]   88 |   return errors;
```

### Additional Info

Here's the file in question, no syntax errors in sight:
https://github.com/danvk/literate-ts/blob/5d04101f895285d838cb4cdccadc8941c86b0771/src/ts-checker.ts#L85

This was quite confusing because there's nothing in the error message indicating that the error is coming from `prettier-plugin-curly`, rather than prettier itself.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #179; fixes #284
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/prettier-plugin-curly/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/prettier-plugin-curly/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Switches AST printing from custom string replacement to the more comprehensive [`@babel/generator`](https://babeljs.io/docs/babel-generator). This fixes at least two known bugs:

* #179: because printing nodes based on source code range doesn't account for leading and trailing comments
* #284: because nodes that contain descendants can clobber those descendants (or vice-versa)

In more detail: printing previously [looped over collected nodes and concatenated text manually](https://github.com/JoshuaKGoldberg/prettier-plugin-curly/blob/50b4dd6d247cb8c7cae90a2250781523e3b9326f/src/preprocess.ts#L66-L70). That works nicely for simple cases but fell apart in the aforementioned edge cases.

Now, the preprocessor modifies the collected AST nodes to ensure their relevant node properties are block statements. Their ranges of the source code are printed anew with `@babel/generator`. This is a kind of "hybrid" approach as I noted in https://github.com/JoshuaKGoldberg/prettier-plugin-curly/pull/309/files#r1527196623:
* It use @babel/generator to print node sections that need re-printing, rather than manual node concatenation
* It avoids re-printing other parts of the AST that are fine as-is

There might be some new introduced edge cases around spacing. See the existing & updated test cases for differences in preprocessed spacing. But, since this is a _pre_-processor that runs before Prettier, I'm hopeful no actual changes make their way to users.

### CODE REVIEW

File: src/preprocess.ts [hunk 1]
```diff
@@ -33,13 +34,14 @@ const getParseOptions = (isJsx: boolean): ParserOptions => ({
 		...(isJsx ? ["jsx" as const] : []),
 		["importAttributes", { deprecatedAssertSyntax: true }],
 	],
+
```
Comment (JoshuaKGoldberg) [inline src/preprocess.ts:37]: This and the changes to `.eslintrc.cjs` in b59729c are an introduced bug. I believe they're due to `@babel/generator` re-printing the AST weirdly. This needs investigation 😕.

Comment (JoshuaKGoldberg) [reply src/preprocess.ts:37]: Added a reduced test case in 66bf77b.

Comment (JoshuaKGoldberg) [reply src/preprocess.ts:11]: Resolved in 122c65f. I ended up going for a hybrid approach:

* Use `@babel/generator` to print node sections that need re-printing
* Avoid re-printing other parts of the AST that are fine as-is
